sc/src/main/java/uk/ac/ebi/atlas/search/GeneSearchService.java

```diff
@@ -66,4 +69,22 @@ public GeneSearchService(GeneSearchDao geneSearchDao) {
             forkJoinPool.shutdown();
         }
     }
+
+    private  Map<String, Map<Integer, List<Integer>>> fetchPreferredKAndMinP (List<String> experimentAccessions, String geneId){
+        Map<String, Map<Integer, List<Integer>>> result = new HashMap<>();
+
+        for (String experimentAccession : experimentAccessions) {
+            Optional<Integer> optional = tsnePlotSettingsService.getExpectedClusters(experimentAccession);
+            if (optional.isPresent()) {
+
+                Integer value = optional.get();
```

Comment:
An assignment is not strictly necessary. You can just do the `.get()` when you pass the parameter below.